src/app/Services/UserService.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace App\Services;
+
+use App\Models\User;
+use App\Models\Post;
+
+class UserService
+{
+    /**
+     * Get the user and posts.
+     *
+     * @return array<string, mixed>
+     */
+    public function getUserProfile(User $user): array
+    {
+        /** @var User $user */
+        $user = auth()->user();
+        $posts = $user->userPosts()->latest()->paginate(4);
+
+        return [
+            'user' => $user,
+            'posts' => $posts,
+        ];
+    }
+
+    /**
+     * Get the user and posts.
+     *
+     * @return array<string, mixed>
+     */
+    public function getUserHomePosts(User $user): array
+    {
+        /** @var User $user */
+        $user = User::with('followings')
+            ->find(auth()->id());
```

Comment:
Same goes here? `$user` is already the param and still you use `User` model and `auth()` helper?

Reply:
I updated my PR po, sir. Thank you!